src/UmbracoFileSystemProviders.Azure/AzureFileSystemComponent.cs

```diff
@@ -0,0 +1,54 @@
+﻿// <copyright file="AzureFileSystemComponent.cs" company="James Jackson-South, Jeavon Leopold, and contributors">
+// Copyright (c) James Jackson-South, Jeavon Leopold, and contributors. All rights reserved.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace Our.Umbraco.FileSystemProviders.Azure
+{
+    using System;
+    using System.Configuration;
+    using global::Umbraco.Core.Components;
+    using global::Umbraco.Core.IO;
+
+    public class AzureFileSystemComponent : IComponent
+    {
+        /// <summary>
+        /// The configuration key for disabling the virtual path provider.
+        /// </summary>
+        private const string DisableVirtualPathProviderKey = Constants.Configuration.DisableVirtualPathProviderKey;
+        private readonly SupportingFileSystems supportingFileSystems;
+
+        public AzureFileSystemComponent(SupportingFileSystems supportingFileSystems)
+        {
+            this.supportingFileSystems = supportingFileSystems;
+        }
+
+        public void Initialize()
+        {
+            bool disable = ConfigurationManager.AppSettings[DisableVirtualPathProviderKey] != null
```

Comment:
This should be at the aggregation root as parameters to the component. Register the component with a factory method where they are injected. IE. `AzureFileSystemComposer` line 40.